### PR TITLE
Add Distro factory guide for DevOps' process automation

### PR DESCRIPTION
**Pull request by mvincerx**
- Adds distro-factory to toctree
- Adds distro-factory in guides to provide DevOps toolchain
- Adds Figure 1 for Distro factory overview.

Signed-off-by: Michael Vincerra <michael.vincerra@intel.com>

### CODE REVIEW

**Comment by mvincerx** [comment]
TCS -- Please hold on review. Work in progress. 

@mbelluzzo, @puneetse, Please provide review and comments as it is, except for `Implementation`. I will replace `Implementation` with more detail after you're satisfied with this as Phase 1.

File: source/clear-linux/guides/maintenance/distro-factory.rst [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  machine for building `mixes`
+
+* Experience using :ref:`mixer <mixer>` to create a |CL|-based distro
+
+* Experience using :ref:`swupd <swupd-guide>`
+
+Overview
+========
+
+Distro factory is divided in two parts: *Content Workflow*; and *Release
```
Comment (mbelluzzo) [inline source/clear-linux/guides/maintenance/distro-factory.rst]: In fact, the maintenance process of Clear Linux is divided in those two parts. Bistro-Factory only handles the 'Release Workflow'. We only mention the Content Workflow as an explanation of the entire process.

File: source/clear-linux/guides/maintenance/distro-factory.rst [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+determine how processes interact. Pipelines can be customized to fit a
+team's requirements. So what do pipelines do?
+
+* `Watcher Pipeline`_ Checks if a new release is necessary by checking |CL| and a content provider, such as Koji.
+* `Release Pipeline`_ Creates new releases when triggered by the Watcher Pipeline
+
+Implementation
+==============
+
+To implement Distro Factory, you should follow a Distro Release Workflow
```
Comment (mbelluzzo) [inline source/clear-linux/guides/maintenance/distro-factory.rst]: This section is looking a bit awkward. I think that the explanation diverged too much and it is not accurate picture of the procedure.

File: source/clear-linux/guides/maintenance/distro-factory.rst (hunk 1 above)
Comment (mvincerx) [reply source/clear-linux/guides/maintenance/distro-factory.rst]: @mbelluzzo. I will rephrase the explanation per your comment. Thanks.

File: source/clear-linux/guides/maintenance/distro-factory.rst (hunk 2 above)
Comment (mvincerx) [reply source/clear-linux/guides/maintenance/distro-factory.rst]: @mbelluzzo. I'll schedule a meeting for us to discuss this.

**Comment by mvincerx** [comment]
For this PR , Phase 1 will stop at Implementation.  For Phase 1, this PR will lay the foundation of the `Content Workflow`, and provide a link to [clr-distro-factory](https://github.com/clearlinux/clr-distro-factory/wiki) for details on `Release Workflow`. 
Phase 2 can capture more details, such as: 
1) Recommended file structure; 
2) clr-distro-factory-config git repo; 
3) Using Jenkins to create jobs for each pipeline.
.. and more.

**Comment by mvincerx** [comment]
@mbelluzzo. Ready for your review. See new title: `architect-lifecycle.rst`

**Comment by mvincerx** [comment]
@mbelluzzo. Unintended capture caused merge conflict. Good now.  View only: `architect-lifecycle.rst`

File: source/clear-linux/guides/maintenance/architect-lifecycle.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+* A repository with software RPM artifacts and a CI/CD system with a |CL|
+  machine for building `mixes`
+
+* Experience using :ref:`mixer <mixer>` to create a |CL|-based distro
+
+* Experience using :ref:`swupd <swupd-guide>` for maintaining the |CL|
+  build environment
+
+* Familiarity with |CL| architecture and reuse of its content in releases
```
Comment (mbelluzzo) [inline source/clear-linux/guides/maintenance/architect-lifecycle.rst]: 👍

Comment (mvincerx) [reply source/clear-linux/guides/maintenance/architect-lifecycle.rst]: Merging now. 

It's worth mentioning again: This is Phase 1 of development on `Distro factory` for output on https://clearlinux.org/documentation/clear-linux/guides. Phases 2-3 will follow.

**Pull request merged**